### ISSUE TITLE
Getting zero accuracy acc = 0

### ISSUE THREAD

**Issue by ftmasadi** [PaddlePaddle/PaddleOCR#9000]
I use the following yaml file, I even spent a few days training it, but every time I change it, it doesn't give me any accuracy.Thank you for helping me, where is the problem with my work?
`Global:
  use_gpu: true
  epoch_num: 5000
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: /content/drive/MyDrive/output
  save_epoch_step: 1
  eval_batch_step:
  - 0
  - 200
  cal_metric_during_train: true
  pretrained_model: /content/PaddleOCR/pretrained_model/arabic_PP-OCRv3_rec_train/best_accuracy
  checkpoints:
  save_inference_dir: null
  use_visualdl: false
  infer_img: /content/PaddleOCR/doc/imgs_words/persian/fa_2.jpg
  character_dict_path: /content/good char.txt
  max_text_length: 120
  infer_mode: false
  use_space_char: true
  save_res_path: /content/drive/MyDrive/output/rec/predicts_svtr_tiny_bn.txt
Optimizer:
  name: AdamW
  beta1: 0.9
  beta2: 0.99
  epsilon: 8.0e-08
  weight_decay: 0.05
  no_weight_decay_name: norm pos_embed
  one_dim_param_no_weight_decay: true
  lr:
    name: Cosine
    learning_rate: 0.00025
    warmup_epoch: 10
Architecture:
  model_type: rec
  algorithm: SVTR
  Transform:
    name: STN_ON
    tps_inputsize:
    - 32
    - 64
    tps_outputsize:
    - 32
    - 100
    num_control_points: 20
    tps_margins:
    - 0.05
    - 0.05
    stn_activation: none
  Backbone:
    name: SVTRNet
    img_size:
    - 32
    - 100
    out_char_num: 120
    out_channels: 192
    patch_merging: Conv
    embed_dim:
    - 64
    - 128
    - 256
    depth:
    - 3
    - 6
    - 3
    num_heads:
    - 2
    - 4
    - 8
    mixer:
    - Local
    - Local
    - Local
    - Local
    - Local
    - Local
    - Global
    - Global
    - Global
    - Global
    - Global
    - Global
    local_mixer:
    - - 7
      - 11
    - - 7
      - 11
    - - 7
      - 11
    last_stage: true
    prenorm: false
  Neck:
    name: SequenceEncoder
    encoder_type: reshape
  Head:
    name: CTCHead
Loss:
  name: CTCLoss
PostProcess:
  name: CTCLabelDecode
Metric:
  name: RecMetric
  main_indicator: acc
Train:
  dataset:
    name: SimpleDataSet
    data_dir: /content/PaddleOCR/train_data/train/
    label_file_list:
    - ./train_data/train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - CTCLabelEncode: null
    - RecResizeImg:
        character_dict_path: null
        image_shape:
        - 3
        - 64
        - 256
        padding: false
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: true
    batch_size_per_card: 64
    drop_last: true
    num_workers: 4
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/val
    label_file_list:
    - ./train_data/val.txt
    ratio_list:
    - 0.1
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - CTCLabelEncode: null
    - RecResizeImg:
        character_dict_path: null
        image_shape:
        - 3
        - 64
        - 256
        padding: false
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 64
    num_workers: 4
profiler_options: null
`

**Comment by ftmasadi** [comment]
This is my result
![2 3](https://user-images.githubusercontent.com/100461197/217292352-4f6f98e5-168a-492f-bc9d-b4960445c20c.png)

**Comment by ftmasadi** [comment]
And some labels of my dataset photos
`00001.tif	ایرانی‌ها چقدر از اینترنت استفاده می‌کنند؟ شناسه خبر: ۹۶۱۰۶ ۲۴ مرداد ۱۳۹۶ - ۱۹:۴۰ بررسی آمارهای
00002.tif	رسمی در سال ۲۰۱۶ گویای آن است که استفاده ایرانیان از در بخش‌های مختلف به
00003.tif	گونه‌ای بوده که ایران با عدد ۳.۳ در رتبه شانزدهم زیرشاخص استفاده قرار دارد که
00004.tif	نسبت به سال ۲۰۱۵ تغییری نداشته است. ۵۵آنلاین : بررسی آمارهای رسمی در سال ۲۰۱۶
00005.tif	گویای آن است که استفاده ایرانیان از در بخش‌های مختلف به گونه‌ای بوده که ایران
00006.tif	با عدد ۳.۳ در رتبه شانزدهم زیرشاخص استفاده قرار دارد که نسبت به سال ۲۰۱۵
00007.tif	تغییری نداشته است. آمارهای رسمی در بخش بخش‌ها و جنبه‌های مختلفی را در برمی‌گیرد که
00008.tif	از این میان یکی از زیرشاخص‌ها مربوط به بحث استفاده کشورها از می‌شود. برای احصای
00009.tif	رتبه کشورهای مختلف در این حوزه تلاش‌های سه گروه اصلی اشخاص، مؤسسات و دولت‌ها را
00010.tif	به‌منظور افزایش ظرفیتشان برای استفاده از اندازه‌گیری می‌کنند. همچنین میزان بهره‌برداری آن‌ها از در فعالیت‌های`

**Comment by ftmasadi** [comment]
And some labels of my dataset photos:
00010.tif	به‌منظور افزایش ظرفیتشان برای استفاده از اندازه‌گیری می‌کنند. همچنین میزان بهره‌برداری آن‌ها از در فعالیت‌های
00011.tif	روزانه در ارتباط با دیگر گروه‌ها مورد اندازه‌گیری قرار می‌گیرد. به طور کلی نیز شاخص

**Comment by ftmasadi** [comment]
@MohieEldinMuha 
I think maybe I'm making a mistake in the shape of my pictures. I saw that your dataset photos are long. What shape did you choose? Thank you for helping me with this problem

**Comment by Topdu** [comment]
`out_char_num` should be set to W//4, W//8 or W//12, and `max_text_length` is set up the same as `out_char_num`.
 
**Long Text Recognition Optimization**

Recognition scenes with most texts containing more than 15 characters will be considered as long text recognition, such as the Chinese sentence level text recognition.

- Resize shape during pre-process

According to experience, Resize shape can be set according to the aspect ratio distribution of the text image during pre-process, i.e., set to the aspect ratio that can cover 90% of the data. For example, when the aspect ratio of 90% of the data is less than 15:1 after collecting the aspect ratio of the dataset, the resize shape should be set as [H, H\*15]. When H is set as 32, it is [32, 480].    

- Use Padding strategy to prevent distortion caused by direct resizing of images

Where there is a text with only few characters in the data, it will cause data distortion and affect training and inference when it is resized directly to a relatively large aspect ratio. Therefore, reserve the aspect ratio first when resizing, then set H as 32. When W<480, pad 0 to W=480., when W>480, resize it to W=480 to retain the original shape of the text image.


#### Adjust the output sequence length

The sequence length of SVTR output is 1\*W/4 by default, i.e., SVTR samples the height of the feature down to 1, and the width down to a quarter of W. However, when we recognize long text, the width is W=480 or higher, and the sequence length is 480//4=120. In practical applications, there are few text images with 120 characters. Obviously, it is unreasonable to use the default structure configuration. Therefore, we should make appropriate adjustments to the SVTR structure, and adjust the value of `out_char_num` in the SVTR structure configuration in the configuration file. According to experience, `out_char_num` is generally set as 1/4, 1/8 or 1/12 of W. 

In this project, W=320, and most of the text includes not more than 40 characters, so we adjust the output sequence length in the configuration file as follows:
```yaml
  Backbone:
    name: SVTRNet
    img_size:
    - 32
    - 320
    out_char_num: 40 # W//4 or W//8 or W//12
    ...
```

In short text recognition scenes, `out_char_num` is generally set as 1/4 of W; in long text recognition scenes, `out_char_num` is generally set as 1/8 or 1/12 of W, which can be set according to actual needs. When W=480, `out_char_num` can be set as 60 or 40. Where there is data with more than 40 characters, `out_char_num` can be set as 60, or 40.

**Comment by Aminfaraji** [comment]
@Topdu 
my samples on dataset is (30,200,3) dimensions. how set hyperparameters on config file?

**Comment by masoudMZB** [comment]
Try to reverse labels. If you read the main paper this model has an LSTM, which looks at the timestamps from to right. as it uses CTC so it is predicting chars from left to right. 
wrong label: اکبر
corrected label : ربکا

**Comment by UserWangZz** [comment]
It has not been updated for a long time. This issue is temporarily closed and can be reopened if necessary.
